daemon/daemon.go

```diff
@@ -1207,7 +1207,7 @@ func (daemon *Daemon) Shutdown(ctx context.Context) error {
 		if ls, err := daemon.Containers(ctx, &types.ContainerListOptions{}); len(ls) != 0 || err != nil {
 			// metrics plugins still need some cleanup
 			daemon.cleanupMetricsPlugins()
-			return nil
+			return err
```

Comment:
Might make sense to include this one in the first commit as well (this line wasn't new, but the change is needed because of the context changes in that commit)

Reply:
Done